lib/puppet/functions/fqdn_rand_string.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# @summary
+#   Generates a random alphanumeric string. Combining the `$fqdn` fact and an
+#   optional seed for repeatable randomness.
+#
+# Optionally, you can specify a character set for the function (defaults to alphanumeric).
+Puppet::Functions.create_function(:fqdn_rand_string) do
+  # @param length The length of the resulting string.
+  # @param charset The character set to use.
+  # @param The seed for repeatable randomness.
+  #
+  # @return [String]
+  #
+  # @example Example Usage:
+  #   fqdn_rand_string(10)
+  #   fqdn_rand_string(10, 'ABCDEF!@$%^')
+  #   fqdn_rand_string(10, '', 'custom seed')
+  dispatch :fqdn_rand_string do
+    param 'Integer[1]', :length
+    optional_param 'String', :charset
+    optional_param 'String', :seed
```

Comment:
It looks like this function use to handle seeds the same way as `fqdn_rotate` in that it accepted zero to many.
_It_ even had [tests](https://github.com/puppetlabs/puppetlabs-stdlib/pull/1341/files#diff-f7e08598996c6cc9c8923e4a4524c628def79565a199d188c000e6e66f4e0dcaR28) for this.

To fully maintain compatibility, you'd have to do something like...
```suggestion
    optional_repeated_param 'String', :seeds
```

Reply:
Good catch, I put this on the side in my mind and forgot about it.  I amended the commits to fix this and am pushing them 1 by 1 to make sure this is backwards compatible. Thanks!